### DISCUSSION TITLE
Is multithreading in the browser with help of WASM here?

### DISCUSSION THREAD

**Discussion by Wandalen** [tokio-rs/tokio#5298] [Q&A]
Maybe someone can answer? Is support is here? Both single-threaded and multithreaded runtimes? Any corner cases? Is it okay to use in production for the web front-end? Any reading or related discussion?

**Comment by Darksonn** [comment]
Tokio has some very limited support for WASM, but it does not include any sort of multi-threading. Additionally, its usually not suitable for browser WASM, where your runtime is the browser instead of Tokio.

**Reply by Wandalen** [reply]
Make sense. Thank you.

**Reply by Wandalen** [reply]
Okay. But what about multi-threading / multiprocessing? How to get it? Just use a pool of threads? Which do you recommend if code should be portable on the web? Maybe you can advice?
@Darksonn

**Reply by Darksonn** [reply]
If you start multiple web workers, then they can run in parallel. I would probably not attempt to put async code inside the web workers — just use ordinary synchronous code. There might be some mechanisms for sharing memory between several web workers, but I don't know how it works, and the most robust is probably a solution where they are isolated.
